### DISCUSSION TITLE
Nuove sezioni (custom post type & custom fields)

### DISCUSSION THREAD

**Discussion by gbvitrano** [opendatasicilia/opendatasicilia.it#42] [Idee/Proposte]
Ricordo male o @picomiles aveva chiesto di inserire un **Calendario eventi** nel nuovo sito?
E' il caso di aggiungere una pagina/sezione **Lavora con Noi** e **Sostieni Open Data Sicilia** ?

**Comment by gabacode** [comment]
Per gli eventi suggerirei di creare un Custom Post Type di tipo evento, eventualmente possiamo anche aggiungere dei Custom Fields, ad esempio:
- Nome Evento
- Luogo Evento (anche coordinate magari)
- Data Evento
- Descrizione evento (contenuto)

La maniera piu facile ed immediata è utilizzando [questo plugin](https://wordpress.org/plugins/custom-post-type-ui/) per i custom types, e [questo](https://wordpress.org/plugins/advanced-custom-fields/) per i custom fields, ma possiamo anche provare a farlo [manualmente](https://www.wpbeginner.com/wp-tutorials/how-to-create-custom-post-types-in-wordpress/)

In questo modo possiamo tenere gli eventi separati da tutti gli altri Articoli.

**Reply by dennisangemi** [reply]
Fantastico! Possiamo fare la stessa cosa per i progetti? Ad esempio:
- Nome progetto 
- Categoria (Es. Dashboard, scraping, API, WebApp, BOT, Mappa, bhu)
- Stato (Es. Attivo, non attivo, in costruzione)
- Descrizione
- Data pubblicazione
- Strumenti utilizzati
- immagine (?)
- contributors (?)

**Reply by dennisangemi** [reply]
Stessa cosa per "Dicono di noi":
- Data pubblicazione
- Titolo
- Testata
- URL

**Comment by dennisangemi** [comment]
> Ricordo male o @picomiles aveva chiesto di inserire un **Calendario eventi** nel nuovo sito? 

Potremmo embeddare un google calendar

**Reply by dennisangemi** [reply]
E capire se si può modificare lo stile così https://www.webdesignerdepot.com/2012/04/integrating-google-calendar-with-your-website/

**Comment by gabacode** [comment]
Ho aggiunto un nuovo custom post type per "Cose Nostre". E' possibile aggiungere nuovi progetti, che supportano nativamente:
- Titolo 
- Contenuto
- Featured Image
- Categoria
- Excerpt 
 
e in piu i custom fields sopra citati:
- Stato
- Data pubblicazione
- Strumenti utilizzati
- Contributors

**Reply by dennisangemi** [reply]
Ciao @gabacode , grazie. Dove pensi possano essere inserite informazioni del tipo `Dashboard, scraping, API, WebApp, BOT, Mappa,` ? Tag? Io avevo scritto `Categoria` ma credo sia inappropriato (sarebbe la stessa categoria del blogpost, no?)

**Reply by dennisangemi** [reply]
A proposito: credo sia necessario inserire un campo per il link al progetto (spesso sono URL di terzo livello)

**Reply by gabacode** [reply]
Potremmo creare un tipo di Tag o Categoria che siano esclusivi al tipo di post. Ho aperto una issue apposita a riguardo #65 
edit: aggiunta custom taxonomy per i post di tipo progetto
![Screenshot_20211219_153622](https://user-images.githubusercontent.com/38985346/146678814-c58f6265-61ac-4579-85c1-36e25923bc35.png)

**Comment by gabacode** [comment]
Avendo adesso implementato alcuni custom types e nuove tassonomie, dovremmo decidere se convertire (previo sacrosanto backup del db) alcuni Post di tipo generico a Post di tipo specifico (ad esempio Eventi, e Progetti), o se estendere i nuovi campi a tutti i tipi di Post o a post di una determinata categoria, continuando ad utilizzare tags e categorie per differenziarli. Io sono piu propenso per la conversione, cosa ne pensate?

**Reply by dennisangemi** [reply]
Sera Gab, innanzitutto grazie sempre. Anche io sono più propenso per la conversione. Eviterei di estendere i nuovi campi a tutti i tipi di post

**Comment by gabacode** [comment]
Come discusso precedentemente nella nostra ultima riunione, al momento ho "replicato" la struttura nativa di WordPress per quanto riguarda url, categorie, e tags, via 3c0b54b

Esempi:
https://dev.opendatasicilia.it/2021/04/22/red-zones-in-sicily-a-story-of-civic-hacking/
https://dev.opendatasicilia.it/category/advocacy
https://dev.opendatasicilia.it/tag/opendata
